src/utils/adagrams_helpers.js

```diff
@@ -0,0 +1,58 @@
+import {
+  LETTER_POOL,
+  NUM_LETTERS,
+  POINT_VALUES
+} from './constants.js'
+
+export const createLetterPool = () => {
+  const letterPool = LETTER_POOL.flatMap(([letter, frequency]) =>
+    Array(frequency).fill(letter)
+  );
+  return letterPool;
+};
+
+export const drawRandomLetter = (pool) => {
+  const randomIndex = Math.floor(Math.random() * pool.length);
+  const letter = pool[randomIndex];
+
+  pool.splice(randomIndex, 1);
+  return letter;
+};
+
+export const getLetterScore = (letter) => {
+  return POINT_VALUES[letter] || 0;
+};
+
+export const updateWinner = (currentData, winningData) => {
+  const { currentWord, currentScore } = currentData;
+  const { winningWord, winningScore } = winningData;
+
+  if (!winningWord || currentScore > winningScore) {
+    return { winningWord: currentWord, winningScore: currentScore };
+  }
+  if (currentScore === winningScore) {
+    return { winningWord: tieBreaker(currentWord, winningWord), winningScore };
+  }
+  return { winningWord, winningScore };
+};
+
+export const tieBreaker = (word, winningWord) => {
```

Comment:
Since `tieBreaker` is only used in this file, I'd be inclined not to expose it. If you find later on that you need to use the method in `adagrams.js`, then you can add the `export` keyword back.

Keeping methods private can prevent unintended access from other parts of your application.

Some teams might have different opinions though so do what your team does in the future.